src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java

```diff
@@ -80,11 +84,60 @@
                 out.add(new Tuple2<>(pair.key(header), pair));
             }
             return out;
-        }).groupByKey();
+        }).groupByKey(parallelism);
 
         return markPairedEnds(keyedPairs, scoringStrategy, finder, header);
     }
 
+    /**
+     * Like <code>groupByKey</code>, but assumes that values are already sorted by key, so no shuffle is needed,
```

Comment:
Thanks for the explanation. In this case, the behaviour (and test) that I've implemented here should be sufficient.